cli/pcluster/config/param_types.py

```diff
@@ -181,7 +181,15 @@ def to_cfn(self):
 
     def get_default_value(self):
         """Get default value from the Param definition if there, None otherwise."""
-        return self.definition.get("default", None)
+        default = self.definition.get("default", None)
+        try:
```

Comment:
I really like the [EAFP](https://docs.python.org/3/glossary.html?highlight=eafp#term-eafp) approach, but probably in this case we could use the [LBYL](https://docs.python.org/3/glossary.html?highlight=eafp#term-lbyl) to test if the default is function.
I say this because most of the time we have a value, not a function, so we can avoid to call the `get_section` and speed up the flow.